File: java/src/main/java/com/ybsid/exercism/medium/IsbnVerifier.java
package com.ybsid.exercism.medium;

/**
 * <a href="https://exercism.org/tracks/java/exercises/isbn-verifier">...</a>
 */
class IsbnVerifier {
    boolean isValid(String stringToVerify) {
        String noDash = stringToVerify.replaceAll("-","");
        if(noDash.length()!=10)
            return false;
        String checkChar = noDash.charAt(9) == 'X' ? "10" : String.valueOf(noDash.charAt(9));
        try {
            int checkDig = Integer.parseInt(checkChar);
            int checkSum = 0;
            for (int i = 0; i <= 8; i++) {
                checkSum += Integer.parseInt(String.valueOf(noDash.charAt(i))) * (10 - i);
            }
            checkSum += checkDig;
            return checkSum % 11 == 0;
        }catch (NumberFormatException e){
            return false;
        }
    }
}

File: python/src/exercises/isbn_verifier/isbn_verifier.py
def is_valid(isbn: str):
    no_dash = isbn.replace("-", "")
    if len(no_dash) != 10:
        return False
    if no_dash[-1] not in "0123456789" and no_dash[-1] != "X":
        return False
    count = 10
    summation = 0
    for char in no_dash[:-1]:
        if char.isdigit():
            summation += int(char) * count
            count -= 1

    if no_dash[-1] == "X":
        summation += 10
    else:
        summation += int(no_dash[-1])

    return summation % 11 == 0


print(is_valid("3-598-21507-X"))
